engine/lib/comments.php

```diff
@@ -231,12 +231,22 @@ function _elgg_comments_container_permissions_override($hook, $type, $return, $p
 function _elgg_comments_permissions_override($hook, $type, $return, $params) {
 	$entity = $params['entity'];
 	$user = $params['user'];
-	
-	if (elgg_instanceof($entity, 'object', 'comment') && $user) {
-		return $entity->getOwnerGUID() == $user->getGUID();
+
+	if (!elgg_instanceof($entity, 'object', 'comment')) {
+		return $return;
 	}
-	
-	return $return;
+
+	if ($entity->owner_guid == $user->guid) {
+	    return true;
+	}
+
+	$commented_on = $entity->getContainerEntity();
+	$container = $commented_on->getContainerEntity();
+	if (elgg_instanceof($container, 'group') && $container->owner_guid == $user->guid) {
+		return true;
+	}
+
+	return false;
```

Comment:
From experience, return `false` from permissions hooks results in all kinds of unwanted side effects. I'd rather we define cases when it should be false and return the default otherwise.